web/src/hashchange.js

```diff
@@ -362,6 +367,9 @@ function do_hashchange_overlay(old_hash) {
     }
 
     if (base === "settings") {
+        if (section === "display-settings" && !overlays.settings_open()) {
+            section = "preferences";
+        }
```

Comment:
Can we do just do this once at the top, just after the `coming_from_overlay` definition, rather than duplicating it here?

Reply:
I tested it and it was working like you said, I'll just push the changes in few minutes.

Reply:
> Can we do just do this once at the top, just after the `coming_from_overlay` definition, rather than duplicating it here?

Pushed the changes made.